### DISCUSSION TITLE
Workflow was terminated after 360 minutes, but config have set timeout-minutes to 600

### DISCUSSION THREAD

**Discussion by EkkoG** [community/community#25700] [Actions]
<aside class="onebox githubactions" data-onebox-src="https://github.com/cielpy/openwrt-dist/actions/runs/999568831">
  <header class="source">

      <a href="https://github.com/cielpy/openwrt-dist/actions/runs/999568831" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">

      <svg width="24" height="24" class="github-icon github-icon-failure" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm8.036-4.024a.75.75 0 00-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 101.06 1.06L12 13.06l2.963 2.964a.75.75 0 001.061-1.06L13.061 12l2.963-2.964a.75.75 0 10-1.06-1.06L12 10.939 9.036 7.976z"></path></svg>

  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/cielpy/openwrt-dist/actions/runs/999568831" target="_blank" rel="noopener nofollow ugc">try to fix task cancled cause 360 min time limit</a>
    </h4>

      <div class="github-info">
        Openwrt Build Bot <span class="github-run-number">#67</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>The log show task have run 600 minutes, but actually it only run 360 minute</p>
<blockquote>
<p>The job running on runner GitHub Actions 2 has exceeded the maximum execution time of 600 minutes.</p>
</blockquote>

**Comment by jsoref** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/cielpy/openwrt-dist/runs/2986062388?check_suite_focus=true">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/cielpy/openwrt-dist/runs/2986062388?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/2119212/181092546-dd5cacac-5f90-472a-b9c6-dead07007f4a.png" class="thumbnail onebox-avatar" width="500" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_500x500.png, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_750x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_1000x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/337b7cc06184984e148c075df20ef009e0e6ecaa_2_10x10.png">

<h3><a href="https://github.com/cielpy/openwrt-dist/runs/2986062388?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">Build software better, together</a></h3>

  <p>GitHub is where people build software. More than 65 million people use GitHub to discover, fork, and contribute to over 200 million projects.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<blockquote>
<p>5h 53m 11s</p>
</blockquote>

**Comment by Simran-B** [comment]
<p>Isn’t it the second job?</p>
<p><img src="https://user-images.githubusercontent.com/7819991/181092548-53114adc-b41a-4f54-8a04-f6e5d6464785.png" alt="image" data-base62-sha1="wri5EDSQMqYuoOAfaGL0DPE8rQR" width="690" height="305"></p>
<p>Anyway, 6 * 60 minutes = 360 minutes. That is the actual limit for a job as per the documentation:</p>
<blockquote>
<ul>
<li>
<strong>Job execution time</strong> - Each job in a workflow can run for up to 6 hours of execution time. If a job reaches this limit, the job is terminated and fails to complete.</li>
</ul>
</blockquote>
<p><a href="https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits" class="onebox" target="_blank" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits</a></p>
<p>The error message</p>
<blockquote>
<p>The job running on runner GitHub Actions 2 has exceeded the maximum execution time of 600 minutes.</p>
</blockquote>
<p>might simply be outdated, not reflecting the actual limit.</p>

**Comment by EkkoG** [comment]
<p><a href="https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes" rel="noopener nofollow ugc">https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes</a> The limit can be changed by config, I have change it to 600, that is why the log show the limit is 600<br>
2021年7月6日 +0800 06:35 Simran via GitHub Support Community <a href="mailto:githubcommunity@discoursemail.com">githubcommunity@discoursemail.com</a>，写道：</p>

**Comment by Simran-B** [comment]
<p>It doesn’t say anything about the hard maximum in the documentation you linked to. Is there any source for it being 600 minutes? Does it only apply to self-hosted runners perhaps?</p>

**Comment by jsoref** [comment, marked as answer]
<p>Yes, you can change the limit. For GitHub hosted runners, the default value corresponds to the maximum value. You can thus only <em>lower</em> the limit, which you’d do to prevent a runaway job from burning through your free or paid minutes.</p>
<p>For self-hosted runners (which you aren’t using), you can set it to a <em>higher</em> (or <em>lower</em>) value. But you aren’t using them, so this is irrelevant to you.</p>
<p>Filed:</p><aside class="onebox githubissue" data-onebox-src="https://github.com/github/docs/issues/7984">
  <header class="source">

      <a href="https://github.com/github/docs/issues/7984" target="_blank" rel="noopener nofollow ugc">github.com/github/docs</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/github/docs/issues/7984" target="_blank" rel="noopener nofollow ugc">jobs.&lt;job_id&gt;.timeout-minutes description is misleading</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2021-07-06" data-time="23:27:52" data-timezone="UTC">11:27PM - 06 Jul 21 UTC</span>
      </div>


      <div class="user">
        <a href="https://github.com/jsoref" target="_blank" rel="noopener nofollow ugc">
          <img alt="jsoref" src="https://user-images.githubusercontent.com/2119212/181092551-34619dcc-a411-4606-b23b-648b2e07c3fb.png" class="onebox-avatar-inline" width="20" height="20">
          jsoref
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          content
        </span>
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          triage
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">### What article on docs.github.com is affected?


https://docs.github.com/en<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">/actions/reference/workflow-syntax-for-github-actions#jobsjob_idtimeout-minutes

### What part(s) of the article would you like to see updated?

&gt; jobs.&lt;job_id&gt;.timeout-minutes
&gt; The maximum number of minutes to let a job run before GitHub automatically cancels it. Default: 360

### Additional information

The text isn't technically wrong, but it's misleading.

Yes, if you use a self-hosted runner, you can set a value that's larger than 360, but if you are using GitHub hosted runners, then this limit is actually a hard limit:

https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits
&gt; Job execution time - Each job in a workflow can run for up to 6 hours of execution time. If a job reaches this limit, the job is terminated and fails to complete.

6*60 = 360

Problems:
1. The text really should explain that 360 is a hard max for GitHub runners explaining that people using GitHub runners can only reduce the limit (to prevent throwing away GitHub minutes, or GitHub billable minutes).
2. Why does one page use 360 minutes while the other list 6 hours? Sure, they're technically equivalent, but by using different units, it makes searching for a value in the other document hard.</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I believe that the confusion you’re experiencing is a likely outcome of the current documentation.</p>

**Comment by Simran-B** [comment]
<p>Is 600 minutes the hard limit for self-hosted runners?  I wonder if higher values might get ignored and the default used instead. There’s another topic where someone says that 360 would be the maximum for self-hosted runners, which may or may not be a bug:</p>
<aside class="quote" data-post="1" data-topic="17121">
  <div class="title">
    <div class="quote-controls"></div>
    
    <a href="https://github.community/t/is-that-possible-to-run-job-which-takes-more-than-6-hours-on-self-hosted-runner/17121">Is that possible to run job which takes more than 6 hours on self-hosted runner?</a> <a class="badge-wrapper  bullet" href="/c/code-to-cloud/github-actions/41"><span class="badge-category-parent-bg" style="background-color: #6a737d;"></span><span class="badge-category-bg" style="background-color: #2188ff;"></span><span style="" data-drop-close="true" class="badge-category clear-badge" title="Discussion and questions about GitHub Actions—automation for all of your development workflows, now with world-class CI/CD.">GitHub Actions</span></a>
  </div>
  <blockquote>
    In my CI/CD setup, one step is to retrain a deep learning model. We setup our own GPU-box as a self-hosted runner to run this step. However, the training takes about one day, which is much longer than the current 6-hours cap GitHub Actions allowed. Is there any way to workaround this limitation? 
  </blockquote>
</aside>

**Reply by IAMDAVID0921** [reply]
based on my experience these couple days, higher value (>6 hrs) will be ignored
